library/src/main/java/com/datatorrent/lib/appdata/gpo/GPOUtils.java

```diff
@@ -178,33 +226,44 @@ public static void setFieldFromJSON(GPOMutable gpo, Type type, String field, JSO
    */
   public static void setFieldFromJSON(GPOMutable gpo, String field, JSONObject jo)
   {
-    Type type = gpo.getFieldDescriptor().getType(field);
+    Object val = getFieldFromJSON(gpo.getFieldDescriptor(), field, jo);
+    gpo.setFieldGeneric(field, val);
+  }
+
+  /**
+   * This method gets the given field from the given {@link JSONObject} and converts the field to an object
+   * of the type specified in the given {@link FieldsDescriptor}.
+   * @param fd The {@link FieldsDescriptor} describing the type of each field.
+   * @param field The field to retrieve from the given {@link JSONObject}.
+   * @param jo The {@link JSONObject} to retrieve a field from.
+   * @return The value of the given field converted to an object of the correct type.
+   */
+  public static Object getFieldFromJSON(FieldsDescriptor fd, String field, JSONObject jo)
+  {
+    Type type = fd.getType(field);
 
-    if(type == Type.BOOLEAN) {
+    if (type == Type.BOOLEAN) {
       Boolean val;
 
       try {
         val = jo.getBoolean(field);
-      }
-      catch(JSONException ex) {
+      } catch (JSONException ex) {
```

Comment:
why not return jo.getBoolean(field); 

This also should be done for LONG, INTEGER, DOUBLE etc